File: 3309-count-prefix-and-suffix-pairs-i/3309-count-prefix-and-suffix-pairs-i.java
class Solution {
    private boolean isPrefixAndSuffix(String str1, String str2) {
        if (str1 == null || str2 == null) {
            return false;
        }
        return str2.startsWith(str1) && str2.endsWith(str1);
    }

    public int countPrefixSuffixPairs(String[] words) {
        int count = 0;
        for (int i = 0; i < words.length; i++) {
            for (int j = 0; j < words.length; j++) {
                if (i >= j) continue;
                if (isPrefixAndSuffix(words[i], words[j])) {
                    count++;
                }
            }
        }
        return count;
    }
}

File: 3309-count-prefix-and-suffix-pairs-i/3309-count-prefix-and-suffix-pairs-i.py
class Solution:
    def countPrefixSuffixPairs(self, words: List[str]) -> int:
        count = 0
        l = len(words)
        for i in range(l):
            for j in range(i + 1, l):
                if words[j].startswith(words[i]) and words[j].endswith(words[i]):
                    count += 1
        return count